blockchain/vm/interpreter.go

```diff
@@ -284,6 +284,22 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte) (ret []byte, err
 		}
 
 		if in.cfg.Debug {
+			// Deterministic gas cost    : opcode and precompiled contract
+			// Non-deterministic gas cost: otherwise
+			if op == STATICCALL {
+				var (
+					addr               = stack.Back(1)
+					targetContractAddr = common.Address(addr.Bytes20())
+					precompiles        = in.evm.GetPrecompiledContractMap(targetContractAddr)
+				)
+				if p := precompiles[targetContractAddr]; p != nil {
+					inOffset, inSize := stack.Back(2), stack.Back(3)
+					args := mem.GetPtr(int64(inOffset.Uint64()), int64(inSize.Uint64()))
+					// TODO: Takes a second return value (computation cost) and makes visible in capture state
+					precompiledGasCost, _ := p.GetRequiredGasAndComputationCost(args)
+					cost = cost - dynamicCost + precompiledGasCost
+				}
+			}
```

Comment:
```suggestion
			if op == CALL || op == CALLCODE || op == DELEGATECALL || op == STATICCALL {
				// In CALL* opcodes, the `cost` includes the three components:
				//
				// 1. The gas cost for the CALL* opcode itself (i.e. operation.constantGas)
				// 2. The dynamic cost incured by memory expansion, account creation, value transfer, etc (see gasCall*())
				// 3. The gas available in the callee, which is the 63/64 of the gas available in the caller (see callGas())
				//    this portion will be partially refunded after callee returns, but we cannot know how much will be refunded at this point.
				//    this portion is stored in evm.callGasTemp (see gasCall*()).
				//
				// In the debug traces, we want to show deterministic components 1 and 2. So we subtract 3 from `cost`.
				cost -= in.evm.callGasTemp

				// If the toAddr is a precompile, add the precompile's gas cost and computation cost in the debug trace.
				// TODO: Add the precompile's computation cost and feed into CaptureState
				var (
					callerAddr  = contract.Address()
					precompiles = in.evm.GetPrecompiledContractMap(callerAddr)
					toAddr      = common.Address(stack.Back(1).Bytes20())
				)
				if p := precompiles[toAddr]; p != nil {
					var inOffset, inSize *uint256.Int
					if op == CALL || op == CALLCODE { // gas, address, value, argsOffset, argsSize, ...
						inOffset, inSize = stack.Back(3), stack.Back(4)
					} else if op == DELEGATECALL || op == STATICCALL { // gas, address, argsOffset, argsSize, ...
						inOffset, inSize = stack.Back(2), stack.Back(3)
					}
					input := mem.GetPtr(int64(inOffset.Uint64()), int64(inSize.Uint64()))
					precompileGas, _ := p.GetRequiredGasAndComputationCost(input)
					cost += precompileGas
				}
```

Rationale
1. we can surgically remove the "gas available to the callee" part thanks to `evm.callGasTemp`
2. The parameter to `GetPrecompiledContractMap` is the caller's address
3. We have to consider all CALL* opcodes

Expected result
```
# Calling the "sha256" precompile
  pc    opcode              gasCost    gasLeft     ccCost     ccLeft
  00471 SUB                       3 9999977934        219  149968849
  00472 DUP2                      3 9999977931        190  149968659
  00473 DUP6                      3 9999977928        165  149968494
  00474 GAS                       2 9999977925        230  149968264
  00475 STATICCALL              172 9999977923      10000  149958264
  00476 ISZERO                    3 9999977751        165  149956999
  00477 DUP1                      3 9999977748        190  149956809
  00478 ISZERO                    3 9999977745        165  149956644
  00479 PUSH2                     3 9999977742        120  149956524

# Calling other contract
  pc    opcode              gasCost    gasLeft     ccCost     ccLeft
  00625 JUMPDEST                  1 9999974789         10  149917204
  00626 POP                       2 9999974788        140  149917064
  00627 GAS                       2 9999974786        230  149916834
  00628 CALL                    100 9999974784       5000  149911834
    00000 PUSH1                     3 9843725080        120  149911714
    00002 PUSH1                     3 9843725077        120  149911594
    00004 MSTORE                   12 9843725074        288  149911306
    00005 CALLVALUE                 2 9843725062        149  149911157
```

Reply:
Looks better. Updated with this. Let's delegate the computation cost addition to another PR.